File: algorithms/1974.minimum-time-to-type-word-using-special-typewriter/solution.rs
use std::cmp::min;

impl Solution {
    pub fn min_time_to_type(word: String) -> i32 {
        let mut prev = 0i32;
        let mut time = 0i32;

        for c in word.chars() {
            let curr = (c as u8 - b'a') as i32;
            let diff = (curr - prev).abs();

            time += min(diff, 26 - diff) + 1;

            prev = curr;
        }

        time
    }
}

File: algorithms/1974.minimum-time-to-type-word-using-special-typewriter/solution.go
func minTimeToType(word string) int {
	prev := 0
	time := 0

	for _, r := range word {
		curr := int(r - 'a')
		diff := int(math.Abs(float64(curr - prev)))

		if diff > 26-diff {
			time += 26 - diff
		} else {
			time += diff
		}
		time += 1

		prev = curr
	}

	return time
}

